### PR TITLE
Simulate shooter

### LINKED ISSUE #7: Add shooter simulation to check code
In order to test the shooter a small physical simulation may be helpful to catch any small bugs in the code.

### PR DESCRIPTION
**Pull request by vichik123**
Leave a comment

### CODE REVIEW

File: src/main/java/frc/robot/command_groups/Shoot.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        );
+    }
+
+//    @Override
+//    public void execute() {
+//        if(shooter.getVelocity() == Shooter.getSetpointVelocity(distance.getAsDouble())){
+//            new FeedToShooter();
+//        }
+//    }
+}
```
Comment (barel13) [inline src/main/java/frc/robot/command_groups/Shoot.java]: You don't need this command for the simulation, delete it

File: src/main/java/frc/robot/subsystems/shooter/commands/FeedToShooter.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package frc.robot.subsystems.shooter.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class FeedToShooter extends CommandBase {
+}
```
Comment (barel13) [inline src/main/java/frc/robot/subsystems/shooter/commands/FeedToShooter.java]: Still not related to the simulation. delete

**Review by barel13 — CHANGES_REQUESTED** [review]

**Review by adamv17 — APPROVED** [review]

**Review by barel13 — APPROVED** [review]

**Comment by katzuv** [comment]
You shouldn't add the shooter code here, only simulation. `simulate-shooter` should be forked off `shooter`, not `main`, otherwise `shooter` has no purpose...

**Comment by katzuv** [comment]
Please change the base of this branch to `shooter` (can be done by editing the PR), then I'll review it.

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 3]
```diff
@@ -102,11 +125,15 @@ public double getVelocity() {
...
-        linearSystemLoop.setNextR(VecBuilder.fill(velocity));
-        linearSystemLoop.correct(VecBuilder.fill(getVelocity()));
+        linearSystemLoop.setNextR(VecBuilder.fill(Units.rotationsToRadians(velocity)));
+        linearSystemLoop.correct(VecBuilder.fill(Units.rotationsToRadians(getVelocity())));
         linearSystemLoop.predict(currentTime - lastTime);
-
-        motor.setVoltage(MathUtil.clamp(linearSystemLoop.getU(0), -NOMINAL_VOLTAGE, NOMINAL_VOLTAGE));
+        if(Robot.isSimulation()){
+            flywheelSim.setInputVoltage(MathUtil.clamp(linearSystemLoop.getU(0), -NOMINAL_VOLTAGE, NOMINAL_VOLTAGE));
+            simCollection.setIntegratedSensorVelocity(unitModel.toTicks100ms(Utils.secondsToMinutes(flywheelSim.getAngularVelocityRPM())));
```
Comment (barel13) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:153]: I think you use the wrong conversion (`secindsToMinutes`)

**Pull request merged**